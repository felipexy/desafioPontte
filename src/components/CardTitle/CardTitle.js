import React from 'react';
import './CardTitle.scss';

const CardTitle = () => {
  return (
    <div className="cardTitle">
      <h2 data-testid="cdTitle" className="cardTitle__cardTitleContent">
        Valor Solicitado
      </h2>
    </div>
  );
};

export default CardTitle;
