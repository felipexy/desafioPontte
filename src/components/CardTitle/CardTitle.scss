.cardTitle {
  border-radius: 15px;
  position: relative;
  width: 70%;
  background:rgb(3, 3, 58);
  margin-right: auto;
  margin-left: auto;
  height: min-content;
  margin-top: -2.8%;
  height: 5vw;
}

.cardTitle::after{
  content: "";
  display: block;
}
  
.cardTitle__cardTitleContent {
  justify-content: center;
  position: relative;
  width: fit-content;
  margin-right: auto;
  margin-left: auto;
  color: white;
  height: min-content;
  margin-top: 1.1%;
  font-size: 2.4vw;
  font-weight: bold;
}
