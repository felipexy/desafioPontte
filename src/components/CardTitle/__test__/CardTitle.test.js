import React from 'react';
import CardTitle from '../CardTitle';
import { render, cleanup, waitFor, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);
describe('Tests for CardTitle component', () => {
  it('Render', async () => {
    const { getByTestId } = render(<CardTitle />);
    const ctnBtt = await waitFor(() => getByTestId('cdTitle'));
    expect(ctnBtt).toHaveTextContent('Valor Solicitado');
  });

  it('Matches snapshot', () => {
    const tree = renderer.create(<CardTitle />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
