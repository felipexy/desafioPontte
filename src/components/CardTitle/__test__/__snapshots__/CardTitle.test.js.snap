// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CardTitle component Matches snapshot 1`] = `
<div
  className="cardTitle"
>
  <h2
    className="cardTitle__cardTitleContent"
    data-testid="cdTitle"
  >
    Valor Solicitado
  </h2>
</div>
`;
